### DISCUSSION TITLE
Parse Ingredient Phrases when enabling Ingredient Amounts

### DISCUSSION THREAD

**Discussion by wcedmisten** [mealie-recipes/mealie#2014] [Feature Request]
![image](https://user-images.githubusercontent.com/4565200/211215874-7204071a-a290-4db4-a541-373b7c204a06.png)

When "Disable Ingredient Amounts" is unselected, the ingredients do not automatically get parsed into quantity, name, unit. It would be a great QoL improvement if this was done automatically. My understanding is that the beta uses https://github.com/mtlynch/ingredient-phrase-tagger to parse the ingredient strings fields, but I'm not sure where this currently gets done. Is that only done on import, or somewhere else?

![image](https://user-images.githubusercontent.com/4565200/211215996-4995ec7f-1efe-456b-ba6c-bf7d5d947ec2.png)

![image](https://user-images.githubusercontent.com/4565200/211215965-898dfc79-3266-430e-bc00-88aa4bd8e763.png)

**Comment by jlssmt** [comment]
@wcedmisten 
after you switch the slider of, you can press the "parse" button.
![image](https://user-images.githubusercontent.com/42897917/213050063-5deba638-98e1-4c39-8a89-d114c8d448aa.png)

**Reply by wcedmisten** [reply]
Ah, great! I figured I might not be looking in the right place. I wonder if this could be in a more obvious location for better UX (or maybe it's just me)

**Comment by wcedmisten** [comment]
Followup on this: it looks like selecting "Disable Ingredient Amounts" and then directly going to the "Parse" flow causes the "Disable Ingredient Amounts" to not be saved. It would be great if the "Parse" flow saved the setting by default.

**Reply by michael-genson** [reply]
This has since been fixed!
